src/compiler/checker.ts

```diff
@@ -2720,26 +2750,31 @@ namespace ts {
             if (!exported || exported === unknownSymbol || exported === moduleSymbol || moduleSymbol.exports!.size === 1 || exported.flags & SymbolFlags.Alias) {
                 return exported;
             }
-            const merged = cloneSymbol(exported);
+            const links = getSymbolLinks(exported);
+            if (links.cjsExportMerged) {

```

Comment:
You supposed that and then came to the unfortunate conclusion that it would not, as there are two callers, each with independent caches of their own, resulting in two separate symbols.

Reply:
oh, now I remember the rest of that unfortunate thread. I think the conclusion was "Wesley will fix this in his PR!", and here we are.